addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitSecuritySystemImpl.java

```diff
@@ -0,0 +1,167 @@
+import org.eclipse.smarthome.core.library.items.StringItem;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.openhab.io.homekit.internal.HomekitAccessoryUpdater;
+import org.openhab.io.homekit.internal.HomekitCharacteristicType;
+import org.openhab.io.homekit.internal.HomekitTaggedItem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beowulfe.hap.HomekitCharacteristicChangeCallback;
+import com.beowulfe.hap.accessories.SecuritySystem;
+import com.beowulfe.hap.accessories.properties.CurrentSecuritySystemState;
+import com.beowulfe.hap.accessories.properties.SecuritySystemAlarmType;
+import com.beowulfe.hap.accessories.properties.TargetSecuritySystemState;
+
+/**
+ * Implements SecuritySystem as a GroupedAccessory made up of multiple items:
+ * <ul>
+ * <li>CurrentSecuritySystemState: String type</li>
+ * <li>TargetSecuritySystemState: String type</li>
+ * </ul>
+ *
+ * @author Cody Cutrer - Initial contribution
+ */
+class HomekitSecuritySystemImpl extends AbstractHomekitAccessoryImpl<GroupItem> implements SecuritySystem {
+
+    @NonNull
+    private StringItem currentSecuritySystemStateItem;
+    private StringItem targetSecuritySystemStateItem;
+
+    private Logger logger = LoggerFactory.getLogger(HomekitSecuritySystemImpl.class);
+
+    public HomekitSecuritySystemImpl(HomekitTaggedItem taggedItem, ItemRegistry itemRegistry,
+            HomekitAccessoryUpdater updater, Map<HomekitCharacteristicType, Item> origCharacteristicItems)
+            throws IncompleteAccessoryException {
+        super(taggedItem, itemRegistry, updater, GroupItem.class);
+
+        HashMap<HomekitCharacteristicType, Item> characteristicItems = new HashMap<>(origCharacteristicItems);
+
+        this.currentSecuritySystemStateItem = Optional
+                .ofNullable(characteristicItems.get(HomekitCharacteristicType.CURRENT_SECURITY_SYSTEM_STATE))
+                .map(m -> (StringItem) m).orElseThrow(() -> new IncompleteAccessoryException(
+                        HomekitCharacteristicType.CURRENT_SECURITY_SYSTEM_STATE));
+
+        this.targetSecuritySystemStateItem = Optional
+                .ofNullable(characteristicItems.get(HomekitCharacteristicType.TARGET_SECURITY_SYSTEM_STATE))
+                .map(m -> (StringItem) m).orElseThrow(
+                        () -> new IncompleteAccessoryException(HomekitCharacteristicType.TARGET_SECURITY_SYSTEM_STATE));
+
+        characteristicItems.entrySet().stream().forEach(entry -> {
+            logger.error("Item {} has unrecognized security system characteristic: {}", entry.getValue().getName(),
+                    entry.getKey().getTag());
+        });
+    }
+
+    @Override
+    public CompletableFuture<CurrentSecuritySystemState> getCurrentSecuritySystemState() {
+        CurrentSecuritySystemState state;
+
+        String stringValue = currentSecuritySystemStateItem.getState().toString();
+        if (stringValue.equalsIgnoreCase("DISARMED")) {
```

Comment:
we can do this in a future PR but these should probably be configurable. See thermostat strings.

Reply:
I know, but I want to wait for item metadata. doing them as global settings is terrible, and I hate how thermostat does it

Reply:
That's a really great point.